port/win/port_win.h

```diff
@@ -70,20 +70,8 @@ extern const bool kDefaultToAdaptiveMutex;
 
 namespace port {
 
-// VS < 2015
-#if defined(_MSC_VER) && (_MSC_VER < 1900)
-
-// VS 15 has snprintf
-#define snprintf _snprintf
-
-#define ROCKSDB_NOEXCEPT
-
-#else // VS >= 2015 or MinGW
-
 #define ROCKSDB_NOEXCEPT noexcept
```

Comment:
It looks like `ROCKSDB_NOEXCEPT` as a define is now obsolete. Recommend either in this PR or a follow-up we simply replace all occurrences with `noexcept`. (Note this is safe/possible because this header is not part of our public API used by others.)

Reply:
Added these changes in this PR.